### PR TITLE
[oneDPL][test] + Add llvm tests for std::abs

### PR DESCRIPTION
**Pull request by SergeyKopienko**
Which functions were tested:
- std::abs

Where these functions were tested:
- on host;
- in Kernel.

Source tests was copied from https://github.com/llvm/llvm-project/tree/main/libcxx/test/std/numerics/c.math

### CODE REVIEW

**Comment by SergeyKopienko** [comment]
@MikeDvorskiy, @akukanov could you please review this PR ?

File: test/xpu_api/numerics/c.math/cmath.pass.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    operator double () { return 0.; }
+};
+
+Ambiguous nearbyint(Ambiguous){ return Ambiguous(); }
+
+template <typename HasDoubleSupportInRuntime, typename HasLongDoubleSupportInCompiletime>
+void test_nearbyint()
+{
+    static_assert((std::is_same<decltype(dpl::nearbyint((float)0)), float>::value), "");
+    IF_DOUBLE_SUPPORT(static_assert((std::is_same<decltype(dpl::nearbyint((bool)0)), double>::value), ""))
```
Comment (MikeDvorskiy) [inline test/xpu_api/numerics/c.math/cmath.pass.cpp:38]: I suggest using just one macro for lines 38-46. Here and everywhere in such blocks of lines.

Comment (SergeyKopienko) [reply test/xpu_api/numerics/c.math/cmath.pass.cpp]: Fixed

File: test/xpu_api/numerics/c.math/cmath.pass.cpp [hunk 2]
```diff
@@ -35,15 +35,19 @@ template <typename HasDoubleSupportInRuntime, typename HasLongDoubleSupportInCom
...
-    IF_DOUBLE_SUPPORT(static_assert((std::is_same<decltype(dpl::nearbyint((int)0)), double>::value), ""))
-    IF_DOUBLE_SUPPORT(static_assert((std::is_same<decltype(dpl::nearbyint((unsigned int)0)), double>::value), ""))
-    IF_DOUBLE_SUPPORT(static_assert((std::is_same<decltype(dpl::nearbyint((long)0)), double>::value), ""))
-    IF_DOUBLE_SUPPORT(static_assert((std::is_same<decltype(dpl::nearbyint((unsigned long)0)), double>::value), ""))
-    IF_DOUBLE_SUPPORT(static_assert((std::is_same<decltype(dpl::nearbyint((long long)0)), double>::value), ""))
-    IF_DOUBLE_SUPPORT(static_assert((std::is_same<decltype(dpl::nearbyint((unsigned long long)0)), double>::value), ""))
-    IF_DOUBLE_SUPPORT(static_assert((std::is_same<decltype(dpl::nearbyint((double)0)), double>::value), ""))
+    auto fnc = []()
+    {
+        static_assert((std::is_same<decltype(dpl::nearbyint((bool)0)), double>::value), "");
```
Comment (MikeDvorskiy) [inline test/xpu_api/numerics/c.math/cmath.pass.cpp:40]: that line number and formatting does match to the corresponded lines an original LLVM test?

**Review by MikeDvorskiy — APPROVED** [review]

**Pull request merged**